lib/workers/bin/run_single_worker.rb

```diff
@@ -60,9 +63,9 @@
   create_options = {:pid => Process.pid}
   runner_options = {}
 
-  if ENV["QUEUE"]
-    create_options[:queue_name] = ENV["QUEUE"]
-    runner_options[:ems_id] = worker_class.ems_id_from_queue_name(ENV["QUEUE"]) if worker_class.respond_to?(:ems_id_from_queue_name)
+  if options[:ems_id]
+    create_options[:queue_name] = options[:ems_id].length == 1 ? "ems_#{options[:ems_id].first}" : options[:ems_id].collect { |id| "ems_#{id}" }
+    runner_options[:ems_id]     = options[:ems_id].length == 1 ? options[:ems_id].first : options[:ems_id].collect { |id| id }
```

Comment:
Can you get rid of your last commit and just turn this `[:ems_id]` line into this?

```ruby
runner_options[:ems_id] = worker_class.ems_id_from_queue_name(create_options[:queue_name]) if worker_class.respond_to?(:ems_id_from_queue_name)
```

The reason I am suggesting this is the code in `ems_id_from_queue_name` should already be tested, where as this script isn't (something I would prefer, but haven't gotten around to refactoring), as well as it seems like these two lines are duplicate work.

Reply:
Scratch the bit about getting rid of your last commit... that was handling allowing `MiqServer` spawning workers with multiple Ems ids and sending it to the script, not the other way around.

Reply:
Agreed with the testing!

Reply:
I *really* don't like any of these lines tbh.

Reply:
I did think the same thing with `ems_id_from_queue_name` its what we want, as this really is just duplicating that, I feel to an extent.

Reply:
Yeah, I'd like to understand how we can move this knowledge and write tests for it.  It's fine here for now. Testing by using foreman/Procfile is not easy to test all the various combinations.